### PR TITLE
[Bluetooth] General fixes & new features

### PR DESCRIPTION
**Pull request by gpotter2**
This PR:
- fixes HCI_ACL_Hdr dissection/building: fix https://github.com/secdev/scapy/issues/368
- fixes ATT_Read_By_Type_Request_128bit detection bug
- add bluetooth link type (now readeable from pcaps/exportable to wireshark)
- fixes EIR_Hdr length computation
- adds BluetoothRFCommSocket socket
- add bluetooth tests
- mark some functions as unavailable on windows

### CODE REVIEW

File: scapy/arch/windows/__init__.py [hunk 1]
```diff
@@ -264,6 +264,16 @@ def test_windump_npcap():
         warning("The installed Windump version does not work with Npcap ! Refer to 'Winpcap/Npcap conflicts' in scapy's doc", True)
     del windump_ok
 
+def _test_windows_bluetooth():
+    """Returns wether or not bluetooth is supported on the computer"""
+    query = exec_query(['Get-WmiObject', 'Win32_NetworkProtocol'],
+                           ['Name'])
+    for protocol in query:
```
Comment (p-l-) [inline scapy/arch/windows/__init__.py]: I don't think you need `.strip()` here. FWIW, I would just:
```Python
def _test_windows_bluetooth():
    """Returns wether or not bluetooth is supported on the computer"""
    return any("bluetooth" in protocol.lower() for protocol in
               exec_query(['Get-WmiObject', 'Win32_NetworkProtocol'], ['Name']))
```

File: scapy/layers/bluetooth.py [hunk 2]
```diff
@@ -394,6 +416,12 @@ class EIR_Hdr(Packet):
         }),
     ]
 
+    def post_build(self, p, pay):
```
Comment (p-l-) [inline scapy/layers/bluetooth.py]: * I don't think it is related to Windows... So maybe it should go in another PR.
* Cannot we use a `*LenField()` for the `len` field value and remove this method?

File: scapy/layers/bluetooth.py [hunk 3]
```diff
@@ -18,6 +20,26 @@
...
+PYBLUEZ = False
+
+if WINDOWS:
+    from scapy.arch.windows import _test_windows_bluetooth
+    if not _test_windows_bluetooth():
+        log_loading.info("Bluetooth is not available on this computer")
+        BT_AVAILABLE = False
+
+try:
+    import bluetooth
```
Comment (p-l-) [inline scapy/layers/bluetooth.py]: I'm not sure to understand this: will we use the `bluetooth` module under other OSes than Windows?

Comment (gpotter2) [reply scapy/layers/bluetooth.py]: the `bluetooth` module, or PyBluez, will be used if available. They use quite a lot of tricks to access the bluetooth sockets on Windows.
It's compulsory for Windows, optional for the other OSes.

File: scapy/arch/windows/__init__.py (hunk 1 above)
Comment (gpotter2) [reply scapy/arch/windows/__init__.py]: way more effective, thanks !

File: scapy/layers/bluetooth.py (hunk 2 above)
Comment (gpotter2) [reply scapy/layers/bluetooth.py]: Right. As the fix became smaller, I'll let it with this PR...
